pytorch_lightning/metrics/classification/auroc.py

```diff
@@ -86,11 +87,12 @@ class AUROC(Metric):
         tensor(0.7778)
 
     """
+
     def __init__(
         self,
         num_classes: Optional[int] = None,
         pos_label: Optional[int] = None,
-        average: Optional[str] = 'macro',
+        average: Optional[AverageMethod] = AverageMethod.MACRO,
```

Comment:
still need to alow strings... I would place it even as default

Reply:
the advantage of our Enum s that they can be natively compared with strings

Reply:
@yuntai you should not focus on the test section but rather here as the API shall not change for the user, we still keep input as a string...